### PR TITLE
feat: refactor FFmpeg soundsource to allow other soundsource to inherit it

### PR DESCRIPTION
**Pull request by acolombier**
This is requirement for the STEM sound source

### CODE REVIEW

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/sources/soundsourceffmpeg.h [hunk 1]
```diff
@@ -131,7 +128,36 @@ class SoundSourceFFmpeg : public SoundSource {
       private:
         AVCodecContext* m_pavCodecContext;
     };
+
+    bool initResampling(
+            audio::ChannelCount* pResampledChannelCount,
+            audio::SampleRate* pResampledSampleRate);
+
+  public:
+    // The following static functions are used by children classes
```
[the remark this answers is not in the document]
Comment (acolombier) [reply src/sources/soundsourceffmpeg.h:137]: I had to make it public so the STEM class, which doesn't inherit FFmpeg directly, but has composition(with the inheritance) can also [open the file](soundsourcestem.cpp:289) to check the the content. I guess I could also make it a friend? What's your thought?

[the remark this answers is not in the document]
Comment (daschuer) [reply src/sources/soundsourceffmpeg.h:137]: No need to become pedantic with it. 
Tha just adjust the comments.

[the remark this answers is not in the document]
Comment (daschuer) [reply src/sources/soundsourceffmpeg.h:137]: No need to become pedantic with it. 
Tha just adjust the comments.

**Review by daschuer — APPROVED** [review]
LGTM, Thank you.

**Comment by daschuer** [comment]
Which one is the next?

**Comment by acolombier** [comment]
That would be [this one](https://github.com/mixxxdj/mixxx/pull/13044)

**Pull request merged**